### PR TITLE
Resolve VCSWP-19762 (Unit Tests for Java SDK)

### PR DESCRIPTION
**Pull request by edgeston**
Closes VCSWP-19762

### CODE REVIEW

File: src/test/java/com/github/freeclimbapi/RecordUtteranceTest.java [hunk 1]
```diff
@@ -16,112 +16,289 @@
...
+ public class RecordUtteranceTest {
     private final RecordUtterance model = new RecordUtterance();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
-
-        Assert.assertEquals("RecordUtterance", model.getCommand());
-    }
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/RecordUtteranceTest.java:40]: This command test should not be removed, please re-add the test back in.

File: src/test/java/com/github/freeclimbapi/RecordUtteranceTest.java [hunk 2]
```diff
@@ -16,112 +16,289 @@
...
+         
+         
+        model.setPlayBeep(false);
+         
+         
+        model.setAutoStart(false);
+         
+         
+        model.setPrivacyMode(false);
+
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/RecordUtteranceTest.java:301]: Formatting also needs to be done here

File: src/test/java/com/github/freeclimbapi/RedirectTest.java [hunk 3]
```diff
@@ -16,47 +16,123 @@
...
+ public class RedirectTest {
     private final Redirect model = new Redirect();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
-
-        Assert.assertEquals("Redirect", model.getCommand());
-    }
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/RedirectTest.java:40]: This command test should not be removed, please re-add the test back in.

File: src/test/java/com/github/freeclimbapi/RedirectTest.java [hunk 4]
```diff
@@ -16,47 +16,123 @@
...
+      String toString1 = test1.toString();
+      Assert.assertTrue(String.class.isInstance(toString1));
     }
 
     @Test
     public void buildTest() throws Exception {
+        
+          
+        model.setActionUrl("TEST_STRING");
+
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/RedirectTest.java:133]: Formatting is needed here as well

File: src/test/java/com/github/freeclimbapi/RejectTest.java [hunk 5]
```diff
@@ -16,47 +16,123 @@
...
+      String toString1 = test1.toString();
+      Assert.assertTrue(String.class.isInstance(toString1));
     }
 
     @Test
     public void buildTest() throws Exception {
+        
+          
+        model.setReason("TEST_STRING");
+
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/RejectTest.java:133]: Formatting is needed here as well

File: src/test/java/com/github/freeclimbapi/RemoveFromConferenceTest.java [hunk 6]
```diff
@@ -16,47 +16,123 @@
...
+ public class RemoveFromConferenceTest {
     private final RemoveFromConference model = new RemoveFromConference();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
-
-        Assert.assertEquals("RemoveFromConference", model.getCommand());
-    }
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/RemoveFromConferenceTest.java:40]: This command test should not be removed, please re-add the test back in.

File: src/test/java/com/github/freeclimbapi/SetListenTest.java [hunk 7]
```diff
@@ -16,60 +16,152 @@
...
+ public class SetListenTest {
     private final SetListen model = new SetListen();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
-
-        Assert.assertEquals("SetListen", model.getCommand());
-    }
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/SetListenTest.java:40]: This command test should not be removed, please re-add the test back in.

File: src/test/java/com/github/freeclimbapi/SetTalkTest.java [hunk 8]
```diff
@@ -16,60 +16,152 @@
...
 
     @Test
     public void buildTest() throws Exception {
+        
+          
+        model.setCallId("TEST_STRING");
+          
+         
+        model.setTalk(false);
+
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/SetTalkTest.java:161]: Formatting needs to be done here

File: src/test/java/com/github/freeclimbapi/SmsTest.java [hunk 9]
```diff
@@ -16,89 +16,216 @@
...
+ public class SmsTest {
     private final Sms model = new Sms();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
-
-        Assert.assertEquals("Sms", model.getCommand());
-    }
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/SmsTest.java:40]: This command test should not be removed, please re-add the test back in.

File: src/test/java/com/github/freeclimbapi/SmsTest.java [hunk 10]
```diff
@@ -16,89 +16,216 @@
...
+        model.setFrom("TEST_STRING");
+          
+         
+          
+        model.setText("TEST_STRING");
+          
+         
+          
+        model.setNotificationUrl("TEST_STRING");
+
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/SmsTest.java:223]: Formatting needs to be done here

File: src/test/java/com/github/freeclimbapi/StartRecordCallTest.java [hunk 11]
```diff
@@ -16,33 +16,32 @@
...
+ public class StartRecordCallTest {
     private final StartRecordCall model = new StartRecordCall();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
-
-        Assert.assertEquals("StartRecordCall", model.getCommand());
-    }
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/StartRecordCallTest.java:40]: This command test should not be removed, please re-add the test back in.

File: src/test/java/com/github/freeclimbapi/UnparkTest.java [hunk 12]
```diff
@@ -16,33 +16,32 @@
...
+ public class UnparkTest {
     private final Unpark model = new Unpark();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
-
-        Assert.assertEquals("Unpark", model.getCommand());
-    }
```
Comment (amahadaya) [inline src/test/java/com/github/freeclimbapi/UnparkTest.java:40]: This command test should not be removed, please re-add the test back in.

**Review by amahadaya — CHANGES_REQUESTED** [review]
I believe tests are good with a few formatting exceptions:

- All properties tests need to be formatted similar to the method tests
- All build tests for PerclCommands needs to be formatted similar to method tests
- Command test for PerclCommands are missing

The comments should address where it needs to be fixed but the properties test is a general formatting fix that needs to be applied.

Good job with the testing autogeneration, files look good.

File: src/test/java/com/github/freeclimbapi/DefaultApiTest.java [hunk 13]
```diff
@@ -52,15 +53,14 @@ public void setup() {
      */
     @Test
     public void buyAPhoneNumberTest() throws ApiException {
-
+        
         BuyIncomingNumberRequest buyIncomingNumberRequest = buyIncomingNumberRequestBuyIncomingNumberRequestTestValue();
-
+        
         IncomingNumberResult response = this.apiInstance.buyAPhoneNumber(buyIncomingNumberRequest);
         // TODO: test validations
```
Comment (katebrune) [inline src/test/java/com/github/freeclimbapi/DefaultApiTest.java:60]: noticed this file had a lot of these comments, was wondering what they mean?

File: src/test/java/com/github/freeclimbapi/DequeueTest.java [hunk 14]
```diff
@@ -16,33 +16,31 @@
...
  * Model tests for Dequeue
  */
-public class DequeueTest {
+ public class DequeueTest {
     private final Dequeue model = new Dequeue();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
```
Comment (katebrune) [inline src/test/java/com/github/freeclimbapi/DequeueTest.java:37]: do we still need this test?

File: src/test/java/com/github/freeclimbapi/EnqueueTest.java [hunk 15]
```diff
@@ -16,89 +16,171 @@
...
  * Model tests for Enqueue
  */
-public class EnqueueTest {
+ public class EnqueueTest {
     private final Enqueue model = new Enqueue();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
```
Comment (katebrune) [inline src/test/java/com/github/freeclimbapi/EnqueueTest.java:37]: https://github.com/FreeClimbAPI/java-sdk/pull/42/files#r1267227627

File: src/test/java/com/github/freeclimbapi/GetSpeechTest.java [hunk 16]
```diff
@@ -27,171 +31,355 @@
...
  * Model tests for GetSpeech
  */
-public class GetSpeechTest {
+ public class GetSpeechTest {
     private final GetSpeech model = new GetSpeech();
-    /**
-     * Test the property 'command'
-     */
-    @Test
-    public void commandTest() {
```
Comment (katebrune) [inline src/test/java/com/github/freeclimbapi/GetSpeechTest.java:39]: https://github.com/FreeClimbAPI/java-sdk/pull/42/files#r1267227627

Comment (edgeston) [reply src/test/java/com/github/freeclimbapi/GetSpeechTest.java:39]: https://github.com/FreeClimbAPI/java-sdk/pull/42#discussion_r1267313581

File: src/test/java/com/github/freeclimbapi/EnqueueTest.java (hunk 15 above)
Comment (edgeston) [reply src/test/java/com/github/freeclimbapi/EnqueueTest.java:37]: https://github.com/FreeClimbAPI/java-sdk/pull/42#discussion_r1267313581

File: src/test/java/com/github/freeclimbapi/DefaultApiTest.java (hunk 13 above)
Comment (edgeston) [reply src/test/java/com/github/freeclimbapi/DefaultApiTest.java]: Hey @katebrune, this is an older generation of the DefaultApiTest. The comments will be changed in future improvements.

**Review by amahadaya — APPROVED** [review]

**Pull request merged**